docs/src/CommutativeAlgebra/rings.md

```diff
@@ -42,36 +45,83 @@ order). The other possible choices are `:deglex` and `:degrevlex`. Gröbner base
 
 ###### Examples
 
-```@repl oscar
-R, (x, y, z) = PolynomialRing(ZZ, ["x", "y", "z"])
-typeof(R)
-typeof(x)
-S, (x, y, z) = PolynomialRing(ZZ, ["x", "y", "z"])
-R === S
+```jldoctest
+julia> R, (x, y, z) = PolynomialRing(ZZ, ["x", "y", "z"])
+(Multivariate Polynomial Ring in x, y, z over Integer Ring, fmpz_mpoly[x, y, z])
+
+julia> typeof(R)
+FmpzMPolyRing
+
+julia> typeof(x)
+fmpz_mpoly
+
+julia> S, (x, y, z) = PolynomialRing(ZZ, ["x", "y", "z"])
+(Multivariate Polynomial Ring in x, y, z over Integer Ring, fmpz_mpoly[x, y, z])
+
+julia> R === S
+true
+
 ```
 
-```@repl oscar
-R1, x = PolynomialRing(QQ, ["x"])
-typeof(x)
-R2, (x,) = PolynomialRing(QQ, ["x"])
-typeof(x)
-R3, x = PolynomialRing(QQ, "x")
-typeof(x)
+```jldoctest
+julia> R1, x = PolynomialRing(QQ, ["x"])
+(Multivariate Polynomial Ring in x over Rational Field, fmpq_mpoly[x])
+
+julia> typeof(x)
+fmpq_poly
+fmpq_mpoly
+Vector{fmpq_mpoly} (alias for Array{fmpq_mpoly, 1})
+
+julia> R2, (x,) = PolynomialRing(QQ, ["x"])
+(Multivariate Polynomial Ring in x over Rational Field, fmpq_mpoly[x])
+
+julia> typeof(x)
+
+julia> R3, x = PolynomialRing(QQ, "x")
+(Univariate Polynomial Ring in x over Rational Field, x)
+
+julia> typeof(x)
+
 ```
 
-```@repl oscar
-V = ["x[1]", "x[2]"]
-T, x = PolynomialRing(GF(3), V)
-x
+```julia
+julia> V = ["x[1]", "x[2]"]
+2-element Vector{String}:
+ "x[1]"
+ "x[2]"
+
+julia> T, x = PolynomialRing(GF(3), V)
+2-element Vector{gfp_mpoly}:
+ x[1]
+ x[2] = PolynomialRing(GF(3), ["x[1]", "x[2]"])
+(Multivariate Polynomial Ring in x[1], x[2] over Galois field with characteristic 3, gfp_mpoly[x[1], x[2]])
+
+julia> x
+
```

Comment:
The output looks messed up. (And the block can be turned into a `jldoctest`.)

Reply:
With some tweaking I think I found a way to avoid the Documenter bug and turn it into a working `jldoctest` anyway. We'll see